cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/DBSource.java

```diff
@@ -124,6 +131,48 @@ private String getJDBCPluginId() {
     return String.format("%s.%s.%s", "source", dbSourceConfig.jdbcPluginType, dbSourceConfig.jdbcPluginName);
   }
 
+  private void ensureValidConnection(Class<? extends Driver> jdbcDriverClass) {
+    try {
+      ensureJDBCDriverIsAvailable(jdbcDriverClass);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("could not ensure JDBCDriver is available", e);
+    }
+    try (Connection connection = createConnection()
+    ) {
+      assert(connection.isValid(0));
+    } catch (SQLException e) {
+      throw new IllegalArgumentException("SQL Exception thrown when trying to connect to driver", e);
```

Comment:
This is especially true because you are not taking any specific action in the catch, just propagating it as an unchecked exception, unlike line 157 below.